From david-b@pacbell.net Thu Feb 15 21:32:36 2007
From: David Brownell <david-b@pacbell.net>
Date: Thu, 15 Feb 2007 21:32:31 -0800
Subject: usbnet: add missing Kconfig for KC2190 cables
To: Greg KH <greg@kroah.com>
Cc: linux-usb-devel@lists.sourceforge.net
Message-ID: <200702152132.31419.david-b@pacbell.net>
Content-Disposition: inline


Hmm, I noticed that support for one of the USB host-to-host cables
never got fully merged ... Kconfig wouldn't show it!  Fixed.

Also, changed the CDC Subset default to 'y' so that more of these
cables will work out-of-the-box.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/net/Kconfig      |    8 ++++++++
 drivers/usb/net/cdc_subset.c |    5 +++++
 2 files changed, 13 insertions(+)

--- gregkh-2.6.orig/drivers/usb/net/Kconfig
+++ gregkh-2.6/drivers/usb/net/Kconfig
@@ -240,6 +240,7 @@ config USB_NET_RNDIS_HOST
 config USB_NET_CDC_SUBSET
 	tristate "Simple USB Network Links (CDC Ethernet subset)"
 	depends on USB_USBNET
+	default y
 	help
 	  This driver module supports USB network devices that can work
 	  without any device-specific information.  Select it if you have
@@ -299,6 +300,13 @@ config USB_EPSON2888
 	  Choose this option to support the usb networking links used
 	  by some sample firmware from Epson.
 
+config USB_KC2190
+	boolean "KT Technology KC2190 based cables (InstaNet)"
+	depends on USB_NET_CDC_SUBSET && EXPERIMENTAL
+	help
+	  Choose this option if you're using a host-to-host cable
+	  with one of these chips.
+
 config USB_NET_ZAURUS
 	tristate "Sharp Zaurus (stock ROMs) and compatible"
 	depends on USB_USBNET
--- gregkh-2.6.orig/drivers/usb/net/cdc_subset.c
+++ gregkh-2.6/drivers/usb/net/cdc_subset.c
@@ -165,6 +165,11 @@ static const struct driver_info	epson288
 #endif	/* CONFIG_USB_EPSON2888 */
 
 
+/*-------------------------------------------------------------------------
+ *
+ * info from Jonathan McDowell <noodles@earth.li>
+ *
+ *-------------------------------------------------------------------------*/
 #ifdef CONFIG_USB_KC2190
 #define HAVE_HARDWARE
 static const struct driver_info kc2190_info = {
